src/panel/wayland-panel-manager.ts

```diff
@@ -1,58 +1,33 @@
-import { registerClass } from "@gi-ts/gobject2";
 import { PanelManager } from "./panel-manager";
+import { spawn_command_line_async } from "@gi-ts/glib2";
 
 const Main = imports.ui.main;
 const PanelBox = Main.layoutManager.panelBox;
-const StatusArea = Main.panel.statusArea;
-const PopupMenu = imports.ui.popupMenu;
-const PanelMenuButton = imports.ui.panelMenu.Button;
-
-const TOP_BAR_DUMB_INDICATOR =
-  "peek-top-bar-on-fullscreen@marcinjahn.com-indicator";
 
 /**
  * On Wayland, making the panel visible is not enough,
  * there is some weird issue that causes the panel to stay invisible,
- * even though it becomes clickable. As a workaround, on Wayland a dumb
- * top bar indicator is added and activated when the panel is supposed
- * to show up. That causes the panel to be visible, at the cost
- * of requiring the user to click outside of the panel to hide it.
+ * even though it becomes clickable. As a workaround, on Wayland a concealed dumb
+ * window with active always on top makes it work as expected.
  */
 export class WaylandPanelManager implements PanelManager {
+  constructor() {
+    spawn_command_line_async(
```

Comment:
all done, except for this one -- I am not sure I am getting what exactly needs to be done, since

1. constructor is basically init() in itself, and
2. we don't have an `X11Manager` anywhere. We have `X11PanelManager`, but the tweak is irelevant to X11

Reply:
Right, I meant `X11PanelManager`, sorry.
In general, I think, and it's not only my opinion, constructors should not do any "big" things outside of initializing the object itself. Creating an external process when invoking a constructor is a "big" thing, and it could be a bit unexpected. That's why I'd prefer for the process to be created in some `init`, or alternatively, there could be some static method in `WaylandPanelManager` that creates the instance and starts the process. Anything other than `new WaylandPanelManager()` will be better in my opinion. When I see `new Something()` I expect the object to be created and initialized with some default internal state. I don't expect some app to be started in another process :)

Having said that, it's alright if we leave it as it is right now, and I can introduce changes in another commit :)